lib/dangermattic/plugins/view_code_pr_checker.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to detect View files in a PR but without having accompanying screenshots.
+  class ViewCodePRChecker < Plugin
+    VIEW_EXTENSIONS_IOS = /(View|Button)\.swift$|\.xib$|\.storyboard$/.freeze
+    VIEW_EXTENSIONS_ANDROID = /(?i)(View|Button)\.(java|kt|xml)$/.freeze
+
+    # Checks if view files have been modified and if a screenshot is included in the pull request body,
+    # displaying a warning if view files have been modified but no screenshot is included.
+    def view_changes_need_screenshots
+      view_files_modified = git.modified_files.any? do |file|
+        VIEW_EXTENSIONS_IOS =~ file || VIEW_EXTENSIONS_ANDROID =~ file
+      end
+      pr_has_screenshots = github.pr_body =~ %r{https?://\S*\.(gif|jpg|jpeg|png|svg){1}}
```

Comment:
I'm not sure what's the point of the `{1}` at the end of this regex? It's basically saying "repeat the previous token exactly 1 time" which should be the same as not having the `{1}` repeater pattern at all 🙃

Reply:
I guess I tried something else in the regex and forgot to remove this 🙃